primitiv/c/graph.cc

```diff
@@ -74,23 +74,40 @@ primitiv_Status primitiv_Node_to_float(
 } PRIMITIV_C_HANDLE_EXCEPTIONS
 
 primitiv_Status primitiv_Node_to_array(
-    const primitiv_Node *node, float *array) try {
-  std::vector<float> v = to_cpp_ptr(node)->to_vector();
-  std::copy(v.begin(), v.end(), array);
+    const primitiv_Node *node, float *array, size_t *array_length) try {
+  const std::vector<float> v = to_cpp_ptr(node)->to_vector();
+  if (array_length) {
+    *array_length = v.size();
+  }
+  if (array) {
+    std::copy(v.begin(), v.end(), array);
```

Comment:
This is dangerous; boundary checking before copying buffer is **always** necessary.
Value of `*array_length` passed from the caller should have the actual size of `array`, but it is never used in this function.

Reply:
I think we can compare `array_length` with `v.size()`, but cannot compare `array_length` with the size of `array` because there is no way to get the size from an array pointer.